tests/functional/constraints/test_constraints.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+from dbt.tests.util import run_dbt, run_dbt_and_capture
+
+models__constraints_column_types_sql = """
+select
+    1000 as int_column,
+    99.99 as float_column,
+    true as bool_column,
+    '2022-01-01'::date as date_column
+"""
+
+models__constraints_incorrect_column_types_sql = """
+select
+    1000 as int_column,
+    99.99 as float_column,
+    true as bool_column,
+    '2022-01-01'::date as date_column
+"""
+
+models__constraints_not_null_sql = """
+select
+    1000 as int_column,
+    99.99 as float_column,
+    true as bool_column,
+    '2022-01-01'::date as date_column
+
+union all
+
+select
+    NULL as int_column,
+    99.99 as float_column,
+    true as bool_column,
+    '2022-01-01'::date as date_column
+"""
+
+models__models_config_yml = """
+version: 2
+models:
+  - name: constraints_column_types
+    description: "Model to test column data type constraints"
+    config:
+      constraints_enabled: true
+    columns:
+      - name: int_column
+        description: "Test for int type"
+        data_type: int
+        constraints: 
+          - unique
+          - not null
+        check: "int_column > 0"
+      - name: float_column
+        description: "Test for int type"
+        data_type: float
+        check: "float_column > 0"
+      - name: bool_column
+        description: "Test for int type"
+        data_type: boolean
+      - name: date_column
+        description: "Test for int type"
+        data_type: date
+
+  - name: constraints_incorrect_column_types
+    description: "Model to test failing column data type constraints"
+    config:
+      constraints_enabled: true
+    columns:
+      - name: int_column
+        description: "Test for int type"
+        data_type: boolean
+      - name: float_column
+        description: "Test for int type"
+        data_type: date
+      - name: bool_column
+        description: "Test for int type"
+        data_type: int
+      - name: date_column
+        description: "Test for int type"
+        data_type: date
+
+  - name: constraints_not_null
+    description: "Model to test failing materialization when a column is NULL"
+    config:
+      constraints_enabled: true
+    columns:
+      - name: int_column
+        description: "Test for int type with some constraints"
+        data_type: int
+        constraints: 
+          - unique
+          - not null
+      - name: float_column
+        description: "Test for int type"
+        data_type: float
+      - name: bool_column
+        description: "Test for int type"
+        data_type: boolean
+      - name: date_column
+        description: "Test for int type"
+        data_type: date
+"""
+
+
+class TestMaterializedWithConstraints:
```

Comment:
Add a rollback to previous table test based on this: https://github.com/dbt-labs/dbt-core/pull/6271/files#diff-ace143195a2ed95e724979d70d36ca44a4177f0876f768795225994d87142da8R139

Reply:
Add a DDL actual == expected test based on this: https://github.com/dbt-labs/dbt-core/pull/6271/files#diff-ace143195a2ed95e724979d70d36ca44a4177f0876f768795225994d87142da8R119

Reply:
Do we really want the DDL test? I feel like any other change to `create_table_as()` would then require updating the tests for something unrelated.

I actually wanted to avoid a test checking the DDL.

Reply:
Based on an internal [slack thread](https://dbt-labs.slack.com/archives/C034EV1RH4Z/p1671465831928679), you don't need the DDL test, but the rollback test remains to be needed

Reply:
Thanks looks great!